aws/resource_aws_sagemaker_endpoint_configuration_test.go

```diff
@@ -252,17 +315,17 @@ func testAccCheckSagemakerEndpointConfigurationExists(n string) resource.TestChe
 
 func testAccSagemakerEndpointConfigurationConfig_Base(rName string) string {
 	return fmt.Sprintf(`
-resource "aws_sagemaker_model" "foo" {
-  name               = %q
-  execution_role_arn = aws_iam_role.foo.arn
+resource "aws_sagemaker_model" "test" {
+  name               = %[1]q
+  execution_role_arn = aws_iam_role.test.arn
 
   primary_container {
     image = "174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"
```

Comment:
@DrFaust92 This container is only available in us-west-2 (so getting failures in GovCloud). Is there a way to look up a container based on the region for this test?

Reply:
cc: @YakDriver

Reply:
Also fails outside the standard commercial test region:

```console
$ AWS_DEFAULT_REGION=us-east-2 make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerEndpointConfiguration_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerEndpointConfiguration_basic -timeout 120m
=== RUN   TestAccAWSSagemakerEndpointConfiguration_basic
=== PAUSE TestAccAWSSagemakerEndpointConfiguration_basic
=== CONT  TestAccAWSSagemakerEndpointConfiguration_basic
    resource_aws_sagemaker_endpoint_configuration_test.go:64: Step 1/2 error: Error running apply: 2020/10/29 15:00:29 [DEBUG] Using modified User-Agent: Terraform/0.12.26 HashiCorp-terraform-exec/0.10.0
        
        Error: error creating Sagemaker model: ValidationException: Region of "174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1" does not match expected region of us-east-2. Cross region ECR image pulls are not allowed.
        	status code: 400, request id: 5b6a1f91-c444-461b-b71a-77cbdd48332d
        
        
--- FAIL: TestAccAWSSagemakerEndpointConfiguration_basic (127.61s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	127.660s
FAIL
GNUmakefile:27: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

Reply:
See https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-algo-docker-registry-paths.html for the value in other regions.

Reply:
I foresee a new data source...

Reply:
yeah, I saw this and thought if I should create a map for that similar to other data source. but as its not the scope of the PR I didn't touch it. I think ill create a subsequent PR to handle this?

Reply:
For short-term, what about a simple switch with hardcoded values? It's not ideal but neither is the Sagemaker API. Any subsequent data source would just be a bunch of hardcoded information.

```suggestion
    image = %[2]q
```